storage/store.go

```diff
@@ -180,16 +193,20 @@ func (e *NotBootstrappedError) Error() string {
 }
 
 // storeRangeIterator is an implementation of rangeIterator which
-// cycles through a store's rangesByKey slice.
+// cycles through a store's rangesByKey btree.
 type storeRangeIterator struct {
 	store     *Store
+	stopper   *util.Stopper
 	remaining int
 	index     int
+	ch        chan *Range // Next range is written to this channel.
```

Comment:
I think we should reimagine the iterator. Creating a go routine and these dueling channels for it seems very awkward. Pushing the stopper through Store's constructor also seems wrong since we're also passing it to `Store.Start()`. For the iterator, I recommend we simply call Ascend newly every time to get a single entry. The iterator does not need to be fast--only lookups in rangesByKey need to be fast. Later, if performance becomes an issue with the iterator, we could always prefetch batches of the next 10 or 100 entries.